### PR TITLE
Add String() method to dependency.Dependency and inner structs

### PR DESCRIPTION
**Pull request by tianon**

### CODE REVIEW

File: dependency/string.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+func (stageSet StageSet) String() string {
+	if len(stageSet.Stages) == 0 {
+		return ""
+	}
+	stages := []string{}
+	for _, stage := range stageSet.Stages {
+		stages = append(stages, stage.String())
+	}
+	return "<" + strings.Join(stages, " ") + ">"
```
Comment (tianon) [inline dependency/string.go]: This one is fun because the parser can't even handle stages yet that I can tell (`Trailing garbage in a Possibility: <`).

Comment (paultag) [inline dependency/string.go]: Hilarious. We should add it.

**Comment by paultag** [comment]
:heart: :heart: :heart:

**Comment by paultag** [comment]
Control files up next!

**Pull request merged**